oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -679,4 +677,117 @@ impl Runtime {
 
         Ok(())
     }
+
+    pub fn new_runtime_proxy(self: Arc<Self>) -> RuntimeProxy {
+        RuntimeProxy {
+            runtime: self.clone(),
+            node_id: self.new_node_reference(),
+        }
+    }
+}
+
+/// A proxy object that binds together a reference to the underlying [`Runtime`] with a single
+/// [`NodeId`].
+///
+/// This can be considered as the interface to the [`Runtime`] that node and pseudo-node
+/// implementations have access to.
+///
+/// Each [`RuntimeProxy`] instance is used by an individual node or pseudo-node instance to
```

Comment:
It would be nice if the name of the type gave more of a hint that it's Node-specific &ndash; maybe `NodeRuntimeProxy` ? 

[Aside: I'd like to sort out the different `Handle` types in the runtime (Issue for discussion to come), so that anything called `Handle` is in a node-relative numbering space.  This `NodeRuntimeProxy` would then help with doing the conversion between node-relative handles and (say) `ChannelId` values.]

Reply:
Perhaps we should keep the `RuntimeProxy` name, but move it under `oak_runtime::node` mod, this may make it clearer that it's node-specific, since at least it would not be alongside `Runtime` itself. I'll leave that for another PR anyways.